### DISCUSSION TITLE
Provider.of with type that is an ancestor of the actual type

### DISCUSSION THREAD

**Discussion by Bonsai11** [rrousselGit/provider#713] [Q&A]
Hi,

I have a Provider which is the ancestor of another Provider, let's call it MyProvider, as they share a big part of their code.
When trying to get the MyProvider using Provider.of\<MyProviderAncestor\>(context, listen: false), the Provider will not be found causing a ProviderNotFoundError.

Is it correct that this Provider is not found even though it technically is of the same type?

The ancestor provider is part of a base which is shared with multiple apps and not to be used directly, but some of the code inside the base also needs to access this provider, and I obviously cannot put the actual provider type inside the base.

If this is indeed the correct behavior, can I somehow make it work differently? Something like Provider.of\<extends MyProviderAncestor\>(context, listen: false) etc.?

Thanks

**Comment by rrousselGit** [comment]
If it's about doing:

```dart
class A implements Interface {}

Provider<A>(create: (ctx) => A());

Provider.of<Interface>(context); // obtains A
```

then no, this is not possible.


But you could do:

```dart
MultiProvider(
  providers: [
    Provider<A>(create: (ctx) => A()),
    Provider<Interface>(create: (ctx) => ctx.read<A>()),
  ],
```

**Reply by Bonsai11** [reply]
I will see if your suggestions works for me.
I meanwhile found a different solution which works good but I'm not sure if its a valid way to go. Something like: 

typedef MyProviderDummy = MyProvider

MyProvider still extends MyProviderAnchestor but in the "base" code I am not using MyProviderAnchestor anymore but instead MyProviderDummy.

This way everything seems to work fine and the only thing I need to do in projects using that base code is to define MyProviderDummy somewhere.

What do you think about that?

**Reply by rrousselGit** [reply]
If that works for you, that'd ok I think
